src/clients/worker/worker.ts

```diff
@@ -370,6 +403,7 @@ export class Worker {
           this.logger.info(`Cancelled step run ${action.stepRunId}`);
         });
         future.cancel('Cancelled by worker');
+        await future.promise;
```

Comment:
not sure about this one. do we still want to wait on the cancelled step run promise?

@grutt

> 🤔 this is a bit of an issue i suppose, we cant handle cancelation because it will likely be invoked in a separate serverless context

Reply:
with the current implementation, we'll make a new request to the worker to cancel the running step. with our grpc model, we send that request to the correct worker which has a ref to the running step run to cancel.

with the webhook worker, the request might get pickedup by a different lambda which will have no reference to the run...

i'd open an issue to track web worker cancellation because this is going to be a tricky one

Reply:
tracking via https://github.com/hatchet-dev/hatchet/issues/584